my-app/src/app/modules/courses-page/courses-page.component.ts

```diff
@@ -16,18 +13,28 @@ import {FilterPipe} from "./core";
   providers: [FilterPipe]
 })
 export class CoursesPageComponent implements OnInit {
-  public courses: Course[] | undefined;
+  public initialCourses: Course[] | undefined;
+  public currentCourses: Course[] | undefined;
   public searchValue: string | undefined;
 
-  constructor(private filter: FilterPipe) {
+  constructor(private filterPipe: FilterPipe,
+              private coursesHandlerService: CoursesHandlerService) {
   }
 
   ngOnInit(): void {
-    this.courses = COURSE_DATA;
+    this.initialCourses = this.coursesHandlerService.getList();
+    this.currentCourses = this.initialCourses;
```

Comment:
You copied a reference here, so I think any changes in `currentCourses` will affect `initialCourses`. I suggest adding `cloneDeep` in `coursesHandlerService.getList`.